scripts/roomba_controller.py

```diff
@@ -0,0 +1,167 @@
+                # Waits for the server to finish performing the action.
+                self._client.wait_for_result()
+                rospy.logwarn("BlockRoomba success: {}".format(self._client.get_result()))
+
+            elif self._state == RoombaControllerStates.hold_position:
+                goal = QuadMoveGoal(movement_type="hold_position", hold_current_position = True)
+                # Sends the goal to the action server.
+                self._client.send_goal(goal)
+                # waits to cancel hold task 
+                rospy.sleep(self._time_to_hold)
+                self._client.cancel_goal()
+                rospy.logwarn("Hold Position Task canceled")
+                self._state = RoombaControllerStates.completed
+
+            elif self._state == RoombaControllerStates.completed:
+                rospy.logwarn("RoombaController was successful")
+
+            else: 
+                rospy.logerr("Roomba Controller is in an invalid state")
+                self._end = True
+                self._state = RoombaControllerStates.invalid_state
+            
+            # state transitioning
+            if not self._client.get_result():
+                if self._state == RoombaControllerStates.failed_task:
+                    self._state == RoombaControllerStates.failed_task_and_recovery
+                else:
+                    self._state == RoombaControllerStates.failed_task
+
+            else:
+                if self._state == RoombaControllerStates.tracking:
+                    if self._tracking_mode:
+                        self._state = RoombaControllerStates.hit
+                    else:
+                        self._state = RoombaControllerStates.block
+
+                elif self._state == RoombaControllerStates.recover_from_success:
+                    if self._holding:
+                        self._state = RoombaControllerStates.hold_position
+                    else:
+                        self._state = RoombaControllerStates.completed
+
+                elif self._state == RoombaControllerStates.hit:
+                    self._state = RoombaControllerStates.recover_from_success
+
+                elif self._state == RoombaControllerStates.block:
+                    self._state = RoombaControllerStates.recover_from_success
+
+                elif self._state == RoombaControllerStates.hold_position:
+                    self._state == RoombaControllerStates.completed
+
+                elif self._state == RoombaControllerStates.completed:
+                    rospy.logwarn("RoombaController was successful")
+
+                else: 
+                    rospy.logerr("Roomba Controller is in an invalid state")
+                    self._end = True
+                    self._state = RoombaControllerStates.invalid_state
+
+            self._status_callback(self._state)    
+
+    def run(self):
+        next_thread = threading.Thread(target=self._next_goal)
+        next_thread.setDaemon(True)
```

Comment:
Is there a good reason this is a daemon thread?

Reply:
I was told to use a daemon thread. so no clue